src/PowerShellEditorServices.Protocol/Server/LanguageServer.cs

```diff
@@ -1225,44 +1226,50 @@ public static string GetDecoratedSymbolName(SymbolReference symbolReference)
             string processId,
             RequestContext<GetRunspaceResponse[]> requestContext)
         {
-            var runspaceResponses = new List<GetRunspaceResponse>();
+            IEnumerable<PSObject> runspaces = null;
 
             if (this.editorSession.PowerShellContext.LocalPowerShellVersion.Version.Major >= 5)
             {
                 if (processId == null) {
                     processId = "current";
                 }
 
-                var isNotCurrentProcess = processId != null && processId != "current";
-
-                var psCommand = new PSCommand();
-
-                if (isNotCurrentProcess) {
-                    psCommand.AddCommand("Enter-PSHostProcess").AddParameter("Id", processId).AddStatement();
-                }
-
-                psCommand.AddCommand("Get-Runspace");
-
-                StringBuilder sb = new StringBuilder();
-                IEnumerable<Runspace> runspaces = await editorSession.PowerShellContext.ExecuteCommandAsync<Runspace>(psCommand, sb);
-                if (runspaces != null)
+                // If the processId is a valid int, we need to run Get-Runspace within that process
+                // otherwise just use the current runspace
+                if (int.TryParse(processId, out int pid))
                 {
-                    foreach (var p in runspaces)
+                    // Create a remote runspace that we will invoke Get-Runspace in
+                    using(var rs = RunspaceFactory.CreateRunspace(new NamedPipeConnectionInfo(pid)))
```

Comment:
Will this work with all supported versions of PowerShell?

Reply:
Good catch... this use to be a runtime check using `Enter-PSHostProcess` but now it's an assembly load check... 

This will work in PowerShell 5.1+

But when I backport I need to ifdef this properly...